diff --git a/node_modules/genkit/lib/tracing.js b/node_modules/genkit/lib/tracing.js
index 987f21503c4f74577884d5f47053de1e7e4871d3..9b400f135b1c312782e259e8751e3a6a9b4372e9 100644
--- a/node_modules/genkit/lib/tracing.js
+++ b/node_modules/genkit/lib/tracing.js
@@ -75,18 +75,25 @@
     let exporter;
     switch (traceStore) {
         case 'firebase':
-            const { FirebaseTraceStore } = await import('@genkit-ai/firebase');
-            exporter = new FirebaseTraceStore();
+            try {
+                const { FirebaseTraceStore } = await import('@genkit-ai/firebase');
+                exporter = new FirebaseTraceStore();
+            } catch (e) {
+                console.warn('To use the Firebase trace store, please install @genkit-ai/firebase');
+            }
             break;
         case 'googleCloud':
             const { GoogleCloudTraceStore } = await import('@genkit-ai/google-cloud');
             exporter = new GoogleCloudTraceStore();
             break;
         case 'dev':
-            const { DevTraceStore } = await import('@genkit-ai/core/lib/dev');
-            exporter = new DevTraceStore();
+            try {
+                const { DevTraceStore } = await import('@genkit-ai/core/lib/dev');
+                exporter = new DevTraceStore();
+            } catch (e) {
+                console.warn('To use the dev trace store, please install @genkit-ai/core/dev');
+            }
             break;
     }
     if (exporter) {
         spanProcessor.addSpanProcessor(new SimpleSpanProcessor(exporter));
